### ISSUE TITLE
Classwork excercise Query

### ISSUE THREAD

**Issue by JibyKrishnakg** [JibyKrishnakg/Classwork_Spring2023#21]
https://github.com/JibyKrishnakg/Classwork_Spring2023/blob/c57af5d0059d6f8e767acadcc6706efbb409430b/test_Classwork_Excercise.py#L5-L21

I am not sure whether my approach to the problem is correct or not.

Could you please guide me on this?

**Comment by dward2** [comment]
Let's start by thinking about the `line_equation` function should do.  You currently have it receiving a parameter `x` and then returning an answer.  What information should be in `x` and what information should be in the answer?

**Comment by JibyKrishnakg** [comment]
My understanding is that 'x' can take any arbitrary value and can be defined by the user and 'y' shall be the output received correponding to the x coordinate on the line created by (x1, y1) and (x2,y2)

**Comment by dward2** [comment]
I see what you are doing.  You are using global variables to define the line.  But, that is now what the exercise is calling for you to do.  The exercise says to develop a Python function that: 
* receives as parameters two tuples `(x1, y1), (x2, y2)`, that represents two (x, y) coordinates on a plane,
* receives a parameter, `x`, that is a new value on the x-coordinate of the above plane

So, you have written a function that does receive the `x` parameter, but it doesn't receive the tuples which define x1, y1, x2, and y2.  How can you modify the `line_equation` function to receive this additional information in the form of parameters?

**Comment by JibyKrishnakg** [comment]
```
def line_equation(x):
    (x1,y1)= input("Enter the first value of the x,y coordinate of the line equation: ")
    (x2,y2)= input("Enter the second value of the x,y coordinate of the line equation: ")
    x= input("Enter the new value of the x coordinate of the line equation: ")
    y = (((y2-y1)/(x2-x1))*(x-x1)) + y2
    return y
```

This is the code when i tried function receiving inputs (x1, y1), (x2, y2)as parameters two tuples.
But when this is called as parameters, how would we able to test them since the user may any values for (x1, y1), (x2, y2).

**Comment by dward2** [comment]
You are correct that this function cannot be tested.  So, you don't want to ask for user input.  You want `line_equation` to receive this info via parameters.  For example, let's say I want to write a function to add two numbers.  How do I have that function receive the two numbers to add?

```python
def add_two_number(x, y):
    answer = x + y
    return answer
```
So, let's look at your `line_equation` function.  It already receives the parameter `x`.  Next, we also want it to receive `x1`, `y1`, `x2`, and `y2` as parameters.  How can we change the definition `line_equation` so that it can receive additional parameters?

**Comment by JibyKrishnakg** [comment]
**Code1: With User inputs**
```
def line_inputs():
    (x1,y1)= input("Enter the first value of the x,y coordinate of the line equation: ")
    (x1)=int(x1)
    (y1)=int(y1)
    (x2,y2)= input("Enter the second value of the x,y coordinate of the line equation: ")
    (x2)=int(x2)
    (y2)=int(y2)
    x= input("Enter the new value of the x coordinate of the line equation: ")
    (x)=int(x)
    return (x,x1,y1,x2,y2)

def line_equation(x,x1,y1,x2,y2):
    y = (((y2-y1)/(x2-x1))*(x-x1)) + y1
    return y

(x,x1,y1,x2,y2)=line_inputs()
y= line_equation(x,x1,y1,x2,y2)
print(y)
```

**Code 2: Without User Inputs**

Without user defined inputs, but shows error since the paramaters are not defined.
```
def line_equation(x,x1,y1,x2,y2):
    y = (((y2-y1)/(x2-x1))*(x-x1)) + y1
    return y

y= line_equation(x,x1,y1,x2,y2)
print(y)
```

**Comment by dward2** [comment]
**Code2** looks great.  This is exactly what `line_equation` should look like.  So, you just need to define values for the parameters before you send them:
```python
def line_equation(x,x1,y1,x2,y2):
    y = (((y2-y1)/(x2-x1))*(x-x1)) + y1
    return y

x1=2 
x2=3 
y1=2 
y2=3 
y= line_equation(x,x1,y1,x2,y2)
print(y)
```
